src/main/java/com/g4s8/mime/MimeTypeOf.java

```diff
@@ -84,20 +82,22 @@ public String subtype() throws IOException {
 
     @Override
     @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
-    public Map<String, List<String>> params() throws IOException {
+    public Map<String, String> params() throws IOException {
         final Matcher match = this.matcher();
         final Matcher param = MimeTypeOf.PTN_PARAM.matcher(this.src);
-        final Map<String, List<String>> map = new HashMap<>(1);
+        final Map<String, String> map = new HashMap<>(1);
         for (int id = match.end(); id < this.src.length(); id = param.end()) {
             param.region(id, this.src.length());
             if (!param.lookingAt()) {
                 throw new IOException("Invalid mime-type params format");
             }
             final String name = param.group(1);
-            if (!map.containsKey(name)) {
-                map.put(name, new LinkedList<String>());
+            if (map.containsKey(name)) {
+                throw new IllegalStateException(
```

Comment:
Here should be checked exception, declared in method signature. You can use `IOException` here or one of successor that suite for error type. If you don't like any you can make new exception class extended from `IOException`. 
BTW if you don't have enough time you can put `@todo` directive ([how to format](https://github.com/yegor256/pdd#how-to-format)) in class javadoc wich will automatically create new github issue.

Reply:
From the Javadoc of IOException I'd say that it's okay for this case. I had a look at the class hierarchy of IOException but couldn't find anything better, so I'll go with it. A custom exception that extends IOException wouldn't have any additional benefits.